src/pcre2_study.c

```diff
@@ -755,11 +755,12 @@ for (;;)
     new ones get added they are properly considered. */
 
     default:
+    PCRE2_DEBUG_UNREACHABLE();
```

Comment:
It should be unreachable, because it is part of a `switch` which is intended to explicitly list every possible opcode. If it is reached, that means the regex bytecode is corrupted, or a new opcode has been added and the `switch` was not updated.

I'm not very sure what the idea of using `PCRE2_ASSERT(0)` would be; could you explain more?